tests/components/lg_soundbar/test_config_flow.py

```diff
@@ -99,35 +132,18 @@ async def test_form_uuid_present_in_both_functions_uuid_q_empty(hass):
     assert result["type"] == "form"
     assert result["errors"] == {}
 
-    mock_uuid_q = MagicMock()
-    mock_name_q = MagicMock()
-
     with patch(
-        "homeassistant.components.lg_soundbar.config_flow.temescal", return_value=Mock()
+        "homeassistant.components.lg_soundbar.config_flow.temescal"
     ) as mock_temescal, patch(
-        "homeassistant.components.lg_soundbar.config_flow.Queue",
-        return_value=MagicMock(),
-    ) as mock_q, patch(
         "homeassistant.components.lg_soundbar.async_setup_entry", return_value=True
     ) as mock_setup_entry:
-        mock_q.side_effect = [mock_uuid_q, mock_name_q]
-        mock_uuid_q.empty.return_value = True
-        mock_uuid_q.get.return_value = "uuid"
-        mock_name_q.get.return_value = "name"
-        tmock = mock_temescal.temescal
-        tmock.return_value = Mock()
-        instance = tmock.return_value
-
-        def temescal_side_effect(addr, port, callback):
-            mac_info = {"msg": "MAC_INFO_DEV", "data": {"s_uuid": "uuid"}}
-            instance.get_mac_info.side_effect = lambda: callback(mac_info)
-            product_info = {"msg": "PRODUCT_INFO", "data": {"s_uuid": "uuid"}}
-            instance.get_product_info.side_effect = lambda: callback(product_info)
-            info = {"msg": "SPK_LIST_VIEW_INFO", "data": {"s_user_name": "name"}}
-            instance.get_info.side_effect = lambda: callback(info)
-            return DEFAULT
-
-        tmock.side_effect = temescal_side_effect
+        setup_mock_temescal(
+            mock_temescal=mock_temescal,
+            mac_info_dev={"s_uuid": "uuid"},
+            product_info={"s_uuid": "uuid"},
+            info={"s_user_name": "name"},
+            delay=2.0,
```

Comment:
Why do we add a delay here? This is not the test that should timeout in the queue. We don't want to have slow tests.

Reply:
This is to test a delay in the callback causing `uuid_q.empty()` at config_flow.py:62 to return True, which may then cause `attr_q.put_nowait(data)` on line 46 to raise Full since the queue maxsize is 1. Without the delay in the test, the item is added to the queue immediately and the check if it's empty always returns False for this test.

Reply:
Changing the callback invocation to use `hass.add_job` instead of a new thread looks like it introduces enough of a delay to enable this test to take the expected path without additional workarounds. Thanks for the tip.